lib/charms/layer/basic.py

```diff
@@ -31,7 +31,12 @@ def bootstrap_charm_deps():
         # if we're using a venv, set it up
         if cfg.get('use_venv'):
             if not os.path.exists(venv):
-                apt_install(['python-virtualenv'])
+                # note that the following two apt_installs *must* be done
```

Comment:
charmhelpers.core.host.lsb_release() will tell you the release, allowing you to explicitly install the correct packages rather than rely on silent failures actually working.